consensus/core/src/leader_scoring.rs

```diff
@@ -63,33 +58,21 @@ impl<'a> ReputationScoreCalculator<'a> {
 
         Self {
             unscored_subdag,
-            committer,
             commit_range,
             scoring_strategy,
             scores_per_authority,
         }
     }
 
     pub(crate) fn calculate(&mut self) -> ReputationScores {
-        let (min_leader_round, max_leader_round) = self.unscored_subdag.get_leader_round_range();
-        let scoring_round_range = self
-            .scoring_strategy
-            .leader_scoring_round_range(min_leader_round, max_leader_round);
-
-        for leader_round in scoring_round_range {
-            for committer in self.committer.get_committers() {
-                tracing::trace!(
-                    "Electing leader for round {leader_round} with committer {committer}"
-                );
-                if let Some(leader_slot) = committer.elect_leader(leader_round) {
-                    tracing::trace!("Calculating score for leader {leader_slot}");
-                    self.add_scores(self.scoring_strategy.calculate_scores_for_leader(
-                        &self.unscored_subdag,
-                        leader_slot,
-                        committer,
-                    ))
-                }
-            }
+        let leaders = self.unscored_subdag.committed_leaders.clone();
+        for leader in leaders.into_iter() {
```

Comment:
nit: you just use `leaders` which implicitly converts to an iterator instead of explicitly do `.into_iter()`